src/main/java/io/github/oliviercailloux/jconfs/location/DistanceDuration.java

```diff
@@ -0,0 +1,164 @@
+ * seconds) for the trajet. The global informations about the trajet (stored in
+ * a Step object) can be get using getGeneralStep method. You can also get all
+ * the steps of the trajet (stored as a List<Step>) using getAllSteps method.
+ * 
+ * @author Anis HAMOUNI & sbourg
+ */
+public class DistanceDuration {
+	private int duration;
+	private int distance;
+	private List<Step> allSteps;
+	private Step generalStep;
+	private Address addressDeparture;
+	private Address addressArrival;
+	private DirectionsDirectionsRoutes trip;
+
+	/**
+	 * 
+	 * Static factory method to build a DistanceDuration instance.
+	 * 
+	 * @param dep   Address
+	 * @param arriv Address
+	 * 
+	 */
+	public static DistanceDuration newDistanceDuration(Address dep, Address arriv) {
+		return new DistanceDuration(dep, arriv);
+	}
+
+	private DistanceDuration(Address dep, Address arriv) {
+		this.duration = 0;
+		this.distance = 0;
+		this.addressDeparture = dep;
+		this.addressArrival = arriv;
+		this.allSteps = new ArrayList<>();
+		this.trip = null;
+		this.generalStep = null;
+	}
+
+	/**
+	 * Return the duration in seconds
+	 * 
+	 * @return duration
+	 * @throws ApiException
+	 */
+	public int getDuration() throws ApiException {
+		if (trip == null)
+			calculateDistanceDuration();
+		return this.duration;
+	}
+
+	/**
+	 * Return the distance in meters
+	 * 
+	 * @return distance
+	 * @throws ApiException
+	 */
+	public int getDistance() throws ApiException {
+		if (trip == null)
+			calculateDistanceDuration();
+		return this.distance;
+	}
+
+	/**
+	 * Return the departure address
+	 * 
+	 * @return addressDeparture
+	 */
+	public Address getDeparture() {
+		return this.addressDeparture;
+	}
+
+	/**
+	 * Return the arrival address
+	 * 
+	 * @return addressArrival
+	 */
+	public Address getArrival() {
+		return this.addressArrival;
+	}
+
+	/**
+	 * Return all the detailed steps of the traject
+	 * 
+	 * @return allSteps
+	 * @throws ApiException
+	 */
+	public List<Step> getAllSteps() throws ApiException {
+		if (trip == null)
+			calculateDistanceDuration();
+
+		String[] tabToParse = this.trip.getLegs().get(0).toString().split("location=\\[");
+		List<String> listToParse = new ArrayList<>(Arrays.asList(tabToParse));
+		listToParse.remove(0);
+		listToParse.remove(0);
+		listToParse.remove(listToParse.size() - 1);
+		for (int i = 0; i < listToParse.size() - 1; i++) {
+			String departureLatitude = listToParse.get(i).split("]")[0].split(",")[1];
+			String departureLongitude = listToParse.get(i).split("]")[0].split(",")[0];
+			String arrivalLatitude = listToParse.get(i + 1).split("]")[0].split(",")[1];
+			String arrivalLongitude = listToParse.get(i + 1).split("]")[0].split(",")[0];
+			Address departureStep = Address.given(null, departureLatitude, departureLongitude);
+			Address arrivalStep = Address.given(null, arrivalLatitude, arrivalLongitude);
+			Step oneNewStep = Step.newStep(departureStep, arrivalStep);
```

Comment:
L’API LocationIQ ne fournit pas les distances entre chaque étape ? Avez-vous vérifié s’il n’y a pas moyen de l’obtenir (sans pour autant multiplier les requêtes) ? Cela semble particulièrement bizarre pour une API qui semble conçue pour un guidage de trajet en voiture.

Reply:
Pour ce point j'ai en effet loupé quelque chose dans l'API. L'API peut effectivement générer les distances et les durées entre chaque 'points' du trajet. Cependant, ces valeurs sont calculées à partir de nodes OpenStreetMap et sont beaucoup plus nombreux que les coordonnées GPS que j'ai parser lors de l'itération précédente (132 couples de coordonnées GPS contre 390 nodes OSM). Il est donc difficile de les rapprocher pour le moment. Voulez-vous que l'on stocke ces nodes OSM dans l'objet Step à la place des coordonnées GPS ?
J'ai cependant enrichi la documentation pour les futurs contribuants au projet qui pourront utiliser les nodes OSM pour générer une carte avec le trajet et avoir les distances et durées associées à chacun des segments.

Reply:
Ok, plaçons ceci en FUTURE alors.